pages/deployments/deploying_to_kubernetes.md.erb

```diff
@@ -0,0 +1,192 @@
+settings ensures only one step runs at a time, effectively creating a deploy
+queue.
+
+Now the pipeline is wired up. Time for the deploy step.
+
+## Setup the Deploy Step
+
+Create a Kubernetes deployment manifest that references the `DOCKER_IMAGE`
+environment variable. `kubectl` does not interpolate environment variables out
+of the box, so you'll need a different solution. The `envsubst` command works
+well for this case. `envsubst` ("environment substitute") replaces environment
+variables from standard input. The result may be piped directly to `kubectl`.
+
+First, You'll need a deployment YML file. Call this file `k8s/deployment.yml`:
+
+```yml
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: tutorial
+  labels:
+    app: tutorial
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: tutorial
+  template:
+    metadata:
+      labels:
+        app: tutorial
+    spec:
+      containers:
+        - name: app
+          image: "${DOCKER_IMAGE}"
+          ports:
+            - containerPort: 3000
+```
+
+<section class="Docs__note">
+  <p>
+    Please refer to the official Kubernetes <a
+    href="https://kubernetes.io/docs/concepts/workloads/controllers/deployment/">deployment
+    documentation</a> for more information. Also note that there are multiple
+    ways to deploy applications to Kubernetes.
+  </p>
+</section>
+
+Now create a short deploy script (`script/buildkite/deploy`):
+
+```bash
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+if [ -z "${DOCKER_IMAGE:-}" ]; then
+  echo "\:boom\: \$DOCKER_IMAGE missing" 1>&2
+  exit 1
+fi
+
+echo '--- \:kubernetes\: Shipping'
+envsubst < k8s/app.yml | kubectl apply -f -
+```
+
+## Perform a Manual Deployment
+
+Open the deployment pipline and click "New Build". Click "Options" and provide the `DOCKER_IMAGE`.
+
+<%= image "manual_deployment.png" %>
+
+Assuming your agents have the required access, then success! :tada:
+
+<%= image "manual_deployment_success.png" %>
+
+## Automate Deployments
+
+[Trigger steps](https://buildkite.com/docs/pipelines/trigger-step) connect
+pipelines. Add a trigger step to the existing test pipeline to trigger the
+deploy pipeline. You may also filter out branches with the `branches` option
+and set environment variables. Remember to set the `DOCKER_IMAGE` environment
+variable.
+
+Add these steps to your existing `.buildkite/pipeline.yml`:
+
+```yml
+  # Add a wait step to only deploy after all steps complete
+  - wait
+
+  - label: ':rocket: Deploy'
+    trigger: kubernetes-tutorial-deploy
+    branches: "master"
+    build:
+      message: "${BUILDKITE_MESSAGE}"
+      commit: "${BUILDKITE_COMMIT}"
+      branch: "${BUILDKITE_BRANCH}"
+      env:
+        # TODO: replace with your Docker image name
+        DOCKER_IMAGE: "asia.gcr.io/buildkite-kubernetes-tutorial/app:${BUILDKITE_BUILD_NUMBER}"
+```
+
+It's magic time. Push some code. :tada: Automated deployments!
+
+<%= image "final_test_pipeline.png" %>
+
+## Notes
```

Comment:
It would be good to rename this section too... not sure what to though. Something about access? Or agent permissions or something?

Also not sure where this section should sit. Maybe just leave it and let me noodle on it for a while, I'll get back to you 🤔